pkg/templates/pdbminavailable/template.go

```diff
@@ -0,0 +1,184 @@
+						{
+							Message: fmt.Sprintf("Failed to retrieve deployments matching the PDB's label selector within namespace %s: %v", pdb.Namespace, err.Error()),
+						},
+					}
+				}
+
+				for _, dl := range deploymentLikes {
+					pdbMinAvailable := value
+					replicas, _ := extract.Replicas(dl)
+					if isPercent {
+						// Calulate the actual value of the MinAvailable with respect to the Replica count if a percentage is set
+						pdbMinAvailable = int(math.Floor(float64(replicas) * (float64(value) / float64(100))))
+					}
+					if replicas <= int32(pdbMinAvailable) {
+						results = append(results, diagnostic.Diagnostic{
+							Message: fmt.Sprintf("Deployment %s has replicas less than or equal to the minimum available replicas set by its PDB.", dl.GetName()),
+						})
+					}
+				}
+				return results
+
+			}, nil
+		}),
+	})
+}
+
+func getDeploymentLikeObjects(lintCtx lintcontext.LintContext, labelSelector labels.Selector, namespace string) ([]k8sutil.Object, error) {
+
+	objectList := make([]k8sutil.Object, 0, len(lintCtx.Objects()))
+
+	for _, obj := range lintCtx.Objects() {
+		// Ensure that only DeploymentLike objects are processed
+		if !objectkinds.IsDeploymentLike(obj.GetK8sObjectName().GroupVersionKind) {
+			continue
+		}
+
+		// Ensure that only DeploymentLikes are in the same namespaces as the PDB
+		if obj.GetK8sObjectName().Namespace != namespace {
+			continue
+		}
+
+		// Build Deployment labelSelector
+		// If there are no selectors on the object, then the PDB won't match the same pods as the Deployment Like
+		selectors, exists := extract.Selector(obj.K8sObject)
+		if exists {
+			// If there are no Replicas set on the Deployment Like, it's not possible to compare to a PDB
+			if _, exists := extract.Replicas(obj.K8sObject); exists {
+				objLabelSelector, err := metaV1.LabelSelectorAsSelector(selectors)
+				if err != nil {
+					return objectList, err
+				}
+				// Find any Deployment Likes with with the same selector as the PDB
+				if labelSelector.String() == objLabelSelector.String() {
+					objectList = append(objectList, obj.K8sObject)
+				}
+			}
+		}
+	}
+
+	return objectList, nil
+}
+
+func getIntOrPercentValueSafelyFromString(intOrStr string) (int, bool, error) {
+
+	var IntOrPercentValue = string("int")
+	intValue, ifString := strconv.Atoi(intOrStr)
+
+	if ifString != nil {
+		IntOrPercentValue = "percent"
+	}
+
+	switch IntOrPercentValue {
+	case "int":
+		return intValue, false, nil
+	case "percent":
+		isPercent := false
+		s := intOrStr
+		if strings.HasSuffix(s, "%") {
+			isPercent = true
+			s = strings.TrimSuffix(s, "%")
+		} else {
+			return 0, false, fmt.Errorf("invalid type: string %q is not a percentage", intOrStr)
+		}
+		v, err := strconv.Atoi(s)
+		if err != nil {
+			return 0, false, fmt.Errorf("invalid value %q: %v", intOrStr, err)
+		}
+		return v, isPercent, nil
+	}
+	return 0, false, fmt.Errorf("invalid type: string %q neither int nor percentage", intOrStr)
+}
```

Comment:
I think it could be simplified
```suggestion
func getIntOrPercentValueSafelyFromString(intOrStr string) (int, bool, error) {
	intValue, ifString := strconv.Atoi(intOrStr)
	if ifString == nil {
		return intValue, false, nil
	}

	s := intOrStr
	if !strings.HasSuffix(s, "%") {
		return 0, false, fmt.Errorf("invalid type: string %q is not a percentage", intOrStr)
	}
	s = strings.TrimSuffix(s, "%")
	v, err := strconv.Atoi(s)
	if err != nil {
		return 0, false, fmt.Errorf("invalid value %q: %v", intOrStr, err)
	}
	return v, true, nil
}
```